### PR TITLE
doc: Add instructions for user added code hosts

### PR DESCRIPTION
**Pull request by ryanslade**
Closes: https://github.com/sourcegraph/sourcegraph/issues/14477

### CODE REVIEW

File: doc/admin/external_service/index.md [hunk 1]
```diff
@@ -12,3 +14,15 @@ Sourcegraph can sync repositories from code hosts and other similar services.
...
+
+- [GitHub](github.md)
+- [GitLab](gitlab.md)
+- [Bitbucket Cloud](bitbucket_cloud.md)
+
+The feature is currently in beta and can be enabled by adding the `AllowUserExternalServicePublic` tag to the user in the Sourcegraph database. An example query to enable this for the user with username `bob` is:
+
+```sql
+UPDATE users SET tags = array_append(tags, 'AllowUserExternalServicePublic') where username='bob';
+```
```
Comment (tsenart) [inline doc/admin/external_service/index.md:28]: Do you think it's OK, conceptually, to put this sort of info in the public product docs? I hadn't thought of the level of detail that was needed, but this seems much more appropriate to the handbook.

Comment (ryanslade) [reply doc/admin/external_service/index.md:28]: In general I wasn't too sure about where to add this as it's currently in private beta. But, it's a product feature not something related to Sourcegraph the company.

Comment (ryanslade) [reply doc/admin/external_service/index.md:28]: Maybe for now we can just leave out the SQL but continue to mention that it's in beta?

Comment (tsenart) [reply doc/admin/external_service/index.md:28]: Yeah, that sounds good to me. We should just on-board all our teammates instead of document how to do it.

**Review by tsenart — APPROVED** [review]

**Comment by tsenart** [comment]
@quinnkeast: Future note: An example of product docs debt here.

1. This content is under `admin/`, but it now relates to users too.
2. It's under `external_service`, which we need to change if we're to consolidate on `code host connections`.

**Pull request merged**